sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastNestedLoopJoinExec.scala

```diff
@@ -205,6 +224,108 @@ case class BroadcastNestedLoopJoinExec(
     }
   }
 
+  case class NotInSubquerySingleColumnOptimizeParams(
+      buildSideHashedRelation: HashedRelation,
+      isNullExists: Boolean,
+      isBuildRowsEmpty: Boolean)
+
+  private def notInSubquerySingleColumnOptimizeEnabled: Boolean = {
+    if (SQLConf.get.notInSubquerySingleColumnOptimizeEnabled && right.output.length == 1) {
+      // BuildSide must be single column, condition must be the following pattern
+      // Or(EqualTo(a, b), IsNull(EqualTo(a, b)))
+      condition.get match {
+        case _ @ Or(
+            _ @ EqualTo(leftAttr: AttributeReference, rightAttr: AttributeReference),
+            _ @ IsNull(_ @ EqualTo(tmpLeft: AttributeReference, tmpRight: AttributeReference)))
+          if leftAttr.semanticEquals(tmpLeft) && rightAttr.semanticEquals(tmpRight) =>
+          notInSubquerySingleColumnOptimizeSetStreamedKey(leftAttr, rightAttr)
+          if (notInSubquerySingleColumnOptimizeStreamedKeyIndex != -1) {
+            true
+          } else {
+            logWarning(s"failed to find notInSubquerySingleColumnOptimizeStreamedKeyIndex," +
```

Comment:
When would this case happen ? Is it a bug ?, ie when would you not be able to find the index ?

If it is not a bug, do we really care about logging it .. it would simplify the control flow if there was one catch all "return false".